Assignment/Assignment.Tests/PingProcessTests.cs

```diff
@@ -59,61 +59,124 @@ public void RunTaskAsync_Success()
     {
         // Do NOT use async/await in this test.
         // Test Sut.RunTaskAsync("localhost");
+        int exitCode = Sut.RunTaskAsync("localhost").Result.ExitCode;
+        Assert.AreEqual<int>(0, exitCode);
     }
 
     [TestMethod]
     public void RunAsync_UsingTaskReturn_Success()
     {
         // Do NOT use async/await in this test.
-        PingResult result = default;
+        PingResult result = Sut.RunAsync("localhost").Result;
         // Test Sut.RunAsync("localhost");
         AssertValidPingOutput(result);
     }
 
     [TestMethod]
-#pragma warning disable CS1998 // Remove this
     async public Task RunAsync_UsingTpl_Success()
     {
         // DO use async/await in this test.
-        PingResult result = default;
+        PingResult result = await Sut.RunAsync("localhost");
 
         // Test Sut.RunAsync("localhost");
         AssertValidPingOutput(result);
     }
-#pragma warning restore CS1998 // Remove this
 
 
     [TestMethod]
     [ExpectedException(typeof(AggregateException))]
     public void RunAsync_UsingTplWithCancellation_CatchAggregateExceptionWrapping()
     {
-        
+        CancellationTokenSource tokenSource = new();
+        Task task = Task.Run( () => Sut.RunAsync("localhost", tokenSource.Token) );
+        tokenSource.Cancel();
+        Assert.IsTrue(tokenSource.IsCancellationRequested);
+        task.Wait();
     }
 
     [TestMethod]
     [ExpectedException(typeof(TaskCanceledException))]
     public void RunAsync_UsingTplWithCancellation_CatchAggregateExceptionWrappingTaskCanceledException()
     {
         // Use exception.Flatten()
+        try
+        {
+            CancellationTokenSource tokenSource = new();
+            Task task = Task.Run(() => Sut.RunAsync("localhost", tokenSource.Token));
+            tokenSource.Cancel();
+            Assert.IsTrue(tokenSource.IsCancellationRequested);
+            task.Wait();
+        }
+        catch (AggregateException ex)
+        {
+            foreach (Exception innerEx in ex.Flatten().InnerExceptions)
+                throw innerEx;
+        }
     }
 
     [TestMethod]
     async public Task RunAsync_MultipleHostAddresses_True()
     {
-        // Pseudo Code - don't trust it!!!
-        string[] hostNames = new string[] { "localhost", "localhost", "localhost", "localhost" };
-        int expectedLineCount = PingOutputLikeExpression.Split(Environment.NewLine).Length*hostNames.Length;
+        string[] hostNames = new string[] { "localhost", "localhost", "localhost", "localhost", "localhost" };
         PingResult result = await Sut.RunAsync(hostNames);
+
+        int expectedLineCount = PingOutputLikeExpression.Split(Environment.NewLine).Length * hostNames.Length;
+        int? lineCount = result.StdOutput?.Split(Environment.NewLine).Length;
+
+        Assert.AreEqual(expectedLineCount, lineCount);

```

Comment:
Use generic AreEqual<T> here.
```suggestion
        Assert.AreEqual<int>(expectedLineCount, lineCount);
```

Reply:
Ah! Missed that! We'll fix that.